### DISCUSSION TITLE
Setting cookie in post request

### DISCUSSION THREAD

**Discussion by RProduction** [alovajs/alova#193] [Q&A]
Hi, I am using alova with AxiosAdapter for my need. I want to ask something.
How to set cookies to be send before request?
Do I set it in the header section?
Seem alova doesn't accept cookie options args when I using typescript.

**Comment by JOU-amjs** [comment]
AxiosAdapter will use axios to request, and you can set the whole params which axios supports.

about cookie, generally, cookies would be added automatically in non-cors request.
if cors request, you must set `withCredentials = true` manually in axios.

so here is a check steps for you.
1. check if wrote cookie to browser.
2. check if set `withCredentials` to true.

to set `withCredentials` in alova, you can follow this.

**set globally**
```js
axios.defaults.withCredentials = true;
```
**set scoped**
```js
alovaInstance.Get('/xxx', {
  // you can set the whole params which axios supports at here.
  withCredentials: true
})
```

**Reply by RProduction** [reply]
Thank you for your answer.
I am using alova for electron application.
I am using it from client side(which need to pass cors and I use it for fetch only), and from back side. The reason why I use it on back side is because I need to set the header of the request dynamically for each post request from back side.

This way I can bypass cors. So maybe even if I set withCredential to true, electron will not automatically set the cookie for me or I misundertand the concept of withCredential.

So what do you think about this?

**Reply by JOU-amjs** [reply]
Based on your description, did you need to pass cookies in nodejs env of electronic?
if so, you can set cookie data by yourself, like this.
```js
const response = await axios.post('http://localhost:3003/auth', {});
axios.defaults.headers.cookie = response.headers['set-cookie']
```

**Reply by JOU-amjs** [reply]
in my cognition, the electronic app is contain both local browser and node env, and use it follow browser rules in browser env, and use it follow nodejs rules in node env, it would be work well.

**Reply by RProduction** [reply]
Wait a minute, we are using alova right? Why set it using axios? Or just use axios instead?

**Reply by JOU-amjs** [reply]
Actually in alova, all requests are dependent on requestAdapter, if you use axios adapter, the alova is use axios to send request, and you can treat alova as a wrapper of axios.
so, the whole params of axios will be supported in alova. you can see [here](https://alova.js.org/tutorial/extension/alova-adapter-axios/#request) for more details about how to set axios params

**Reply by RProduction** [reply]
Oh okay, I have successfully sent a request with cookie. Thank you very much.

**Reply by JOU-amjs** [reply]
Congratulations! 🎉😁
